### PR TITLE
feat: update hugo/netflify config

### PR DESCRIPTION
**Pull request by JTorreG**
### Proposed changes
Update Netlify and Hugo configuration to publish documentation directly from this repository.
Update the F5-Hugo theme to the latest version

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [CONTRIBUTING](https://github.com/nginxinc/nginx-service-mesh/blob/main/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that all unit tests pass after adding my changes
- [X] I have updated necessary documentation
- [X] I have rebased my branch onto main
- [X ] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

File: docs/config/development/config.toml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+baseURL = "https://docs-dev.nginx.com/nginx-service-mesh"
+title = "NGINX Service Mesh Development Docs "
```
Comment (sjberman) [inline docs/config/development/config.toml:2]: extra whitespace at the end

File: docs/config/preview/config.toml [hunk 2]
```diff
@@ -0,0 +1,4 @@
+baseURL = "https://mesh-preview.netlify.app/nginx-service-mesh"
+title = "NGINX Service Mesh Preview Docs "
```
Comment (sjberman) [inline docs/config/preview/config.toml:2]: extra whitespace at the end

File: docs/config/production/config.toml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+baseURL = "https://docs.nginx.com/nginx-service-mesh"
+title = "NGINX Service Mesh Docs "
```
Comment (sjberman) [inline docs/config/production/config.toml:2]: extra whitespace at the end

File: docs/config/staging/config.toml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+baseURL = "https://docs-staging.nginx.com/nginx-service-mesh"
+title = "STAGING -- NGINX Service Mesh Docs"
+publishDir = "public/nginx-service-mesh"
+canonifyURLs = false
```
Comment (sjberman) [inline docs/config/staging/config.toml:4]: missing newline at the end of this file

**Review by sjberman — APPROVED** [review]

**Comment by jbyers19** [comment]
@Jcahilltorre, how do the dev, preview, and staging deployments work?

**Comment by Jcahilltorre** [comment]
> @Jcahilltorre, how do the dev, preview, and staging deployments work?



> @Jcahilltorre, how do the dev, preview, and staging deployments work?

same as in the previous repo. Netlify can build different versions of the docs from different branches. That's how we populate our staging and development preview sites

**Review by jbyers19 — APPROVED** [review]

**Pull request merged**